test/dynamo/test_dynamo.py

```diff
@@ -769,6 +769,35 @@ def foo(x):
 
     self.assertEqual(expected, actual.cpu())
 
+class DynamoDynamicShapeTest(unittest.TestCase):
+
+  def simple_add(self, a, b):
+    c = a + b
+    return c
```

Comment:
we need a more real-world case. Can you try the resnet18 and uses different batch size?

Reply:
Good point, updated! Any suggestions on what metrics we want to assert on? With this change, now we directly call `optimized_mod` when shape (batch size, in this case) changes. However, we still call `extract_graph_helper` in `optimized_mod` which will trigger the compile.